ui/src/components/project/TableItem.vue

```diff
@@ -271,9 +274,12 @@ export default class extends Vue {
         sources: this.sources,
         save: async (attribute: ReferenceAttributeFormData) => {
           const loading = this.$buefy.loading.open({})
-          await this.$store.dispatch('attributes/createReference', { table: this.table, attribute })
-          loading.close()
-          modal.close()
+          try {
+            await this.$store.dispatch('attributes/createReference', { table: this.table, attribute })
+            modal.close()
+          } finally {
+            loading.close()
+          }
```

Comment:
Will `Promise` API be a little cleaner then `try..catch`?

```suggestion
          await this.$store.dispatch('attributes/createReference', { table: this.table, attribute })
            .then(() => modal.close())
            .finally(() => loading.close())
```

Reply:
Hum... I guess it's a matter of taste? I generally try not to mix `async/await` with the Promise API if I can avoid it. But can change if you have an argument for it ;)

Reply:
No strong arguments. I merely thought it a little more appealing... visually

Reply:
Ok, then I'll keep it that way if you don't mind